orc/src/main/java/org/apache/iceberg/orc/OrcFileAppender.java

```diff
@@ -99,9 +104,34 @@ public Metrics metrics() {
 
   @Override
   public long length() {
-    Preconditions.checkState(isClosed,
-        "Cannot return length while appending to an open file.");
-    return file.toInputFile().getLength();
+    if (isClosed) {
+      return file.toInputFile().getLength();
+    }
+    if (this.treeWriter != null) {
+      if (batch.size > 0) {
+        try {
+          writer.addRowBatch(batch);
+          batch.reset();
```

Comment:
`length` should not modify the writer.

Reply:
Removed this, but if didn't flush here, I was a little concerned that it will deviate a little from the actual size, although it may not be small.

Reply:
I want to get the byte size of the `batch `here, but I haven't found the method yet.

Reply:
Because we check RollToNewFile when we process every record, so if you flush batch here, does it mean that each piece of data will be flushed during processing?

Reply:
Because we check RollToNewFile when we process every record, so if you flush batch here, does it mean that each piece of data will be flushed during processing?

Reply:
Yes, it is poor wirte. It would be better to get the `batch` length here and add it to `length result`.

Reply:
Deviating from the actual size is okay. While the file is open, this is an estimate only.